Source/Core/VideoBackends/OGL/PostProcessing.cpp

```diff
@@ -12,270 +12,953 @@
-					switch (it.second.m_integer_values.size())
-					{
-					case 1:
-						glUniform1i(m_uniform_bindings[it.first], it.second.m_integer_values[0]);
-					break;
-					case 2:
-						glUniform2i(m_uniform_bindings[it.first],
-								it.second.m_integer_values[0],
-						            it.second.m_integer_values[1]);
-					break;
-					case 3:
-						glUniform3i(m_uniform_bindings[it.first],
-								it.second.m_integer_values[0],
-								it.second.m_integer_values[1],
-						            it.second.m_integer_values[2]);
-					break;
-					case 4:
-						glUniform4i(m_uniform_bindings[it.first],
-								it.second.m_integer_values[0],
-								it.second.m_integer_values[1],
-								it.second.m_integer_values[2],
-						            it.second.m_integer_values[3]);
-					break;
-					}
+
+			case POST_PROCESSING_INPUT_TYPE_COLOR_BUFFER:
+				m_last_pass_uses_color_buffer = true;
 				break;
-				case PostProcessingShaderConfiguration::ConfigurationOption::OptionType::OPTION_FLOAT:
-					switch (it.second.m_float_values.size())
-					{
-					case 1:
-						glUniform1f(m_uniform_bindings[it.first], it.second.m_float_values[0]);
-					break;
-					case 2:
-						glUniform2f(m_uniform_bindings[it.first],
-								it.second.m_float_values[0],
-						            it.second.m_float_values[1]);
-					break;
-					case 3:
-						glUniform3f(m_uniform_bindings[it.first],
-								it.second.m_float_values[0],
-								it.second.m_float_values[1],
-						            it.second.m_float_values[2]);
-					break;
-					case 4:
-						glUniform4f(m_uniform_bindings[it.first],
-								it.second.m_float_values[0],
-								it.second.m_float_values[1],
-								it.second.m_float_values[2],
-						            it.second.m_float_values[3]);
-					break;
-					}
+
+			default:
+				break;
+			}
+		}
+	}
+}
+
+void PostProcessingShader::Draw(OGLPostProcessor* parent,
+								const TargetRectangle& dst_rect, const TargetSize& dst_size, GLuint dst_texture,
+								const TargetRectangle& src_rect, const TargetSize& src_size, GLuint src_texture,
+								GLuint src_depth_texture, int src_layer)
+{
+	_dbg_assert_(VIDEO, m_ready && m_internal_size == src_size);
+	OpenGL_BindAttributelessVAO();
+
+	// Determine whether we can skip the final copy by writing directly to the output texture, if the last pass is not scaled.
+	bool skip_final_copy = !IsLastPassScaled() && (dst_texture != src_texture || !m_last_pass_uses_color_buffer);
+
+	// If the last pass is not at full scale, we can't skip the copy.
+	if (m_passes[m_last_pass_index].output_size != src_size)
+		skip_final_copy = false;
+
+	// Draw each pass.
+	PostProcessor::InputTextureSizeArray input_sizes;
+	TargetRectangle output_rect = {};
+	TargetSize output_size;
+	for (size_t pass_index = 0; pass_index < m_passes.size(); pass_index++)
+	{
+		const RenderPassData& pass = m_passes[pass_index];
+		bool is_last_pass = (pass_index == m_last_pass_index);
+
```

Comment:
all the passes are executed even if they are disabled, i think this lines are missing here 
if (!pass.enabled)
			continue;